filedescriptor/src/unix.rs

```diff
@@ -146,6 +165,35 @@ impl OwnedHandle {
         }
     }
 
+    #[inline]
+    pub(crate) unsafe fn dup2_impl<F: AsRawFileDescriptor>(
```

Comment:
For a bit of complexity, I think `dup3` is a linux only interface, so I suspect that you'll need to do something like:

```rust

#[cfg(target_os = "linux")]
{
   // try to use dup3 here
}
// fall back to non-atomic dup2 here
```

Reply:
It's already in the unix mod. This is cfg toggled in lib so should never get called when running in windows.

Reply:
If you want me to build a windows version I'm not sure how to deal with cloexec management. What would I need to do for this?

Reply:
what I meant was, the only unix with `dup3` is linux.  macos and freebsd don't have it, so you'll need to explicitly check for linux to call `dup3`.

Reply:
Ah apologies, completely misread this. Will change now.

Reply:
Done